### PR TITLE
ApplicationInstallation CR: add ApplicationVersion field in the status

### PR DESCRIPTION
**Pull request by vgramer**
**What does this PR do / Why do we need it**:

this pr is the first part of  #8384. it updates the ApplicationInstallation status to store that applicationVersion used to install /uninstall the application. This is needed because if an applicationDefinition is updated and an applicationVersion is removed from it, we need to keep track to be able to uninstall the application

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml [hunk 1]
```diff
@@ -73,6 +73,244 @@ spec:
...
+                        description: Define the valued that can be override for the
+                          installation
+                        items:
+                          properties:
+                            displayName:
+                              description: DisplayName is visible in the UI
+                              type: string
+                            helpText:
+                              description: HelpText is visible in the UI next to the
+                                filed
```
Comment (hdurand0710) [inline charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml]: s/filed/field

File: charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml [hunk 2]
```diff
@@ -73,6 +73,244 @@ spec:
...
+                          type: object
+                        type: array
+                      method:
+                        description: Method used to install the application
+                        enum:
+                        - helm
+                        type: string
+                      source:
+                        description: Defined how the source of the application (e.g
+                          Helm chart) are retrieved
```
Comment (hdurand0710) [inline charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml]: s/are/is

File: charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml [hunk 3]
```diff
@@ -73,6 +73,244 @@ spec:
...
+                        properties:
+                          git:
+                            description: Get application to install form a Git repository
```
Comment (hdurand0710) [inline charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml]: s/form/from

File: charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml [hunk 4]
```diff
@@ -73,6 +73,244 @@ spec:
...
+                          satisfy for the application to be installed. Wildcards are
+                          allowed  e.g. "2.18.*" Range are allowed e.g.">= 2.16, <
+                          2.18.0"
+                        type: string
+                    type: object
+                  template:
+                    description: Template defines how application is installed (source
+                      provenance, Method...)
+                    properties:
+                      formSpec:
```
Comment (hdurand0710) [inline charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml:101]: Is it `formSpec` or `fromSpec` ?

Comment (vgramer) [reply charts/kubermatic-operator/crd/k8c.io/apps.kubermatic.k8c.io_applicationinstallations.yaml:101]: it's `formSpec` like in the addons its define the specification of the form in UI to specify values

**Pull request merged**